src/Compilers/CSharp/Portable/Binder/Binder_Lookup.cs

```diff
@@ -177,34 +183,72 @@ protected void LookupMembersInternal(LookupResult result, NamespaceOrTypeSymbol
         }
 
 #nullable enable
-        private void LookupExtensionMembersInSingleBinder(LookupResult result, TypeSymbol receiverType,
-            string name, int arity, ConsList<TypeSymbol>? basesBeingResolved, LookupOptions options,
-            Binder originalBinder, ref CompoundUseSiteInfo<AssemblySymbol> useSiteInfo)
+        private void LookupAllExtensionMembersInSingleBinder(LookupResult result, TypeSymbol receiverType,
+            string? name, int arity, LookupOptions options, Binder originalBinder,
+            ref CompoundUseSiteInfo<AssemblySymbol> classicExtensionUseSiteInfo)
         {
-            Debug.Assert(name is not null);
+            var singleLookupResults = ArrayBuilder<SingleLookupResult>.GetInstance();
+            EnumerateAllExtensionMembersInSingleBinder(singleLookupResults, receiverType, name, arity, options, originalBinder, ref classicExtensionUseSiteInfo);
+            foreach (var singleLookupResult in singleLookupResults)
+            {
+                result.MergeEqual(singleLookupResult);
+            }
 
-            var extensions = ArrayBuilder<NamedTypeSymbol>.GetInstance();
+            singleLookupResults.Free();
+        }
 
-            Debug.Assert(!receiverType.IsDynamic());
-            if (!receiverType.IsErrorType())
+        internal void EnumerateAllExtensionMembersInSingleBinder(ArrayBuilder<SingleLookupResult> result,
+            TypeSymbol receiverType, string? name, int arity, LookupOptions options, Binder originalBinder,
+            ref CompoundUseSiteInfo<AssemblySymbol> classicExtensionUseSiteInfo)
+        {
+            // 1. Collect new extension members
+            PooledHashSet<MethodSymbol>? implementationsToShadow = null;
+            if (!receiverType.IsErrorType() && !receiverType.IsDynamic())
             {
-                this.GetExtensionDeclarations(extensions, originalBinder);
+                var extensionDeclarations = ArrayBuilder<NamedTypeSymbol>.GetInstance();
+                this.GetExtensionDeclarations(extensionDeclarations, originalBinder);
+
+                foreach (NamedTypeSymbol extensionDeclaration in extensionDeclarations)
+                {
+                    var candidates = name is null ? extensionDeclaration.GetMembers() : extensionDeclaration.GetMembers(name);
+
+                    foreach (var candidate in candidates)
+                    {
+                        SingleLookupResult resultOfThisMember = originalBinder.CheckViability(candidate, arity, options, null, diagnose: true, useSiteInfo: ref classicExtensionUseSiteInfo);
+                        result.Add(resultOfThisMember);
+
+                        if (!candidate.IsStatic)
```

Comment:
>if (!candidate.IsStatic) [](http://example.com/codeflow?start=24&length=24)

It looks like this condition is redundant (taking the comment for the line 222 into consideration)